pkg/platform/provider/baremetal/cluster/create.go

```diff
@@ -892,6 +911,9 @@ func (p *Provider) EnsureGalaxy(ctx context.Context, c *v1.Cluster) error {
 	if c.Status.Phase == platformv1.ClusterUpscaling {
 		return nil
 	}
+	if c.Cluster.Spec.Features.EnableCilium {
+		return nil
+	}
```

Comment:
the check L914-L916 is no good,  if EnableCilium is true, you should put EnsureGalaxy to skipConditions at high level